Assets/EngyneCreations/Trello/Scripts/TrelloBoardSettings.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace Trello
+{
+	public class TrelloBoardSettings : ScriptableObject
+	{
+		[Header("API Key and Token needed for Interacting with Trello")]
+		public string APIKey;
+		public string APIToken;
+
+		private static TrelloBoardSettings _instance;
+
+		public static TrelloBoardSettings Instance
+		{
+			get
+			{
+				LoadSettings();
+				return _instance;
+			}
+		}
+
+		public static void LoadSettings()
+		{
+			_instance = FindOrCreateInstance();
+		}
+
+		private static TrelloBoardSettings FindOrCreateInstance()
+		{
+			TrelloBoardSettings instance = null;
+			instance = Resources.Load<TrelloBoardSettings>("TrelloBoard");
+			instance = instance ? instance : Resources.LoadAll<TrelloBoardSettings>(string.Empty).FirstOrDefault();
+			instance = instance ? instance : CreateAndSave<TrelloBoardSettings>();
+
+			if (instance == null)
+			{
+				throw new Exception("Could not find or create settings for Trello API");
+			}
+
+			return instance;
+		}
+		
+		private static T CreateAndSave<T>() where T : ScriptableObject
+		{
+			T instance = CreateInstance<T>();
+#if UNITY_EDITOR
```

Comment:
I find this kind of strange to put this here. 
Isn't this whole component supposed to be run only in the Unity_editor?

Reply:
For now, but since the TrelloApi is not reliant on the Editor and we could be using `TrelloApi` for handling bug reports and stuff like in-game, having that Macro just helps for future proofing